### DISCUSSION TITLE
autocancel with useQuery / fetch on components mount - following react practice

### DISCUSSION THREAD

**Discussion by LocusDagraca** [pocketbase/pocketbase#6139] [Q&A]
I'm a heavy pocketbase user, i love it. 
I just end up on a small edge case:

I have 2 seperate components on the same page, both fetch the same pocketbase query on mount (like a page refresh)
-> one of the two will get a autocancel.

How would you @ganigeorgiev deal with such situation?

To clarify more:
- to follow react practice, both components have their own data fetching on mount/init.
- they do different things but rely on one similar request, like a getUser()
- pocketbase request to get user will be triggered by both, on page mount
- those requests are wrapped in a useQuery hook from tanstack (like a useIsUserTrial, useIsUserPro etc, and they use my getUser under the hood)

Current fix:
- i added a optionnal prop to add a "requestKey: null" for getUser, so custom hooks on mount like useQuery will use this from now

I just wonder if its the right way to avoid the auto cancel.

Or would you centralize requests in a kind of way?

Thanks and have great end of year.

**Comment by pnmcosta** [comment]
I'm not a React expert, but I would indeed centralize the single request in a Provider and pass the result in a Context, both your components should be able to acquire it.

**Comment by ReinforceZwei** [comment]
You may consider disabling auto-cancellation.
```js
const pocketbase = new PocketBase(PB_URL)
pocketbase.autoCancellation(false)
```

React query library usually come with query caching. If two query with same parameters are triggered, the second query should return the cached data instead of making second request.
I think TanStack Query you have to manually specify a `queryKey` for the cache to work. See https://tanstack.com/query/latest/docs/framework/react/guides/query-keys
